packages/flutter_driver/lib/src/driver/web_driver.dart

```diff
@@ -0,0 +1,270 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:webdriver/sync_io.dart' as sync;
+import 'package:webdriver/support/async.dart';
+
+import '../common/error.dart';
+import '../common/message.dart';
+import 'driver.dart';
+import 'timeline.dart';
+import 'web_driver_config.dart';
+
+/// An implementation of the Flutter Driver using the WebDriver.
+class WebFlutterDriver extends FlutterDriver {
+  /// Creates a driver that uses a connection provided by the given
+  /// [_connection] and [_browserName].
+  @visibleForTesting
+  WebFlutterDriver.connectedTo(
+      this._connection,
+      this._browserName);
+
+  WebFlutterDriver._connect(this._connection, this._browserName);
+
+  final FlutterWebConnection _connection;
+  final String _browserName;
+  DateTime _startTime;
+
+  /// Creates a driver that uses a connection provided by the given
+  /// [hostUrl].
+  static Future<FlutterDriver> connectWeb({String hostUrl}) async {
+    hostUrl ??= Platform.environment['VM_SERVICE_URL'];
+    final String browserName = Platform.environment['BROWSER_NAME'];
+    final Map<String, dynamic> settings = <String, dynamic>{
+      'browser-name': browserName,
+      'browser-dimension': Platform.environment['BROWSER_DIMENSION'],
+      'headless': Platform.environment['HEADLESS']?.toLowerCase() == 'true',
+      'selenium-port': Platform.environment['SELENIUM_PORT'],
+    };
+    final FlutterWebConnection connection = await FlutterWebConnection.connect
+      (hostUrl, settings);
+    return WebFlutterDriver._connect(connection, browserName);
+  }
+
+  @override
+  Future<Map<String, dynamic>> sendCommand(Command command) async {
+    Map<String, dynamic> response;
+    final Map<String, String> serialized = command.serialize();
+    try {
+      final dynamic data = await _connection.sendCommand('window.\$flutterDriver(\'${jsonEncode(serialized)}\')', command.timeout);
+      response = data != null ? json.decode(data) : <String, dynamic>{};
+    } catch (error, stackTrace) {
+      throw DriverError('Failed to respond to $command due to remote error\n : \$flutterDriver(\'${jsonEncode(serialized)}\')',
+          error,
+          stackTrace
+      );
+    }
+    if (response['isError'] == true)
+      throw DriverError('Error in Flutter application: ${response['response']}');
+    return response['response'];
+  }
+
+  @override
+  Future<void> close() => _connection.close();
+
+  @override
+  Future<void> forceGC() async {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<List<Map<String, Object>>> getVmFlags() async {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<List<int>> screenshot() async {
+    // HACK: this artificial delay here is to deal with a race between the
```

Comment:
Rather than duplicating this comment, it would be better to make it a doc comment on the base class